pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -80,27 +80,32 @@ let
   cudaArchitecturesString = strings.concatStringsSep ";" cudaArchitectures;
   minArch =
     let
-      minArch' = builtins.head (builtins.sort builtins.lessThan cudaArchitectures);
+      minArch' = builtins.head (builtins.sort strings.versionOlder cudaArchitectures);
     in
-    # If this fails some day, something must've changed and we should re-validate our assumptions
-    assert builtins.stringLength minArch' == 2;
```

Comment:
why was this removed?

Reply:
Because the assumption is incorrect, assuming NVIDIA keeps going with the same conventions they've used for the last decade. Come Blackwell, we should see the arrival of compute capability 10.0, which would translate to a `minArch'` of `"100"`, which is longer than 2.

I don't see the check providing much value, though if you'd like I can change the check to `>= 2` instead of `== 2`.

Reply:
gotcha gotcha, makes sense!